activation/activation.go

```diff
@@ -639,28 +644,38 @@ func (b *Builder) createAtx(ctx context.Context) (*types.ActivationTx, error) {
 	//                           WAITING FOR POET        DEADLINE
 	//                               PROOFS
 	// NiPoST must be ready before start of the next poet round.
-	pubEpoch := b.challenge.PublishEpoch()
+	pubEpoch := challenge.PublishEpoch()
+	poetRoundStart := b.layerClock.LayerToTime((pubEpoch - 1).FirstLayer()).Add(b.poetCfg.PhaseShift)
 	nextPoetRoundStart := b.layerClock.LayerToTime(pubEpoch.FirstLayer()).Add(b.poetCfg.PhaseShift)
 	poetRoundEnd := nextPoetRoundStart.Add(-b.poetCfg.CycleGap)
 	poetProofDeadline := poetRoundEnd.Add(b.poetCfg.GracePeriod)
 
 	b.log.With().Info("building NIPost",
-		log.Stringer("poet round start", nextPoetRoundStart),
-		log.Stringer("pub_epoch", pubEpoch),
-		log.Time("deadline time", poetProofDeadline),
+		log.Time("poet round start", poetRoundStart),
+		log.Time("next poet round start", nextPoetRoundStart),
+		log.Time("poet proof deadline", poetProofDeadline),
+		log.FieldNamed("current epoch", b.currentEpoch()),
```

Comment:
:s/FieldNamed/Stringer

Reply:
It would result in `"epoch_id"` instead of `"current epoch"`. I prefer the latter to differentiate between 3 epochs logged here.

Reply:
What's the benefit? `EpochID` implements `log.LoggableField`. Using `log.FieldNamed` I get the same serialization as in other logs, except with a different _key_. By using `Stringer` it could have a different serialization.

Reply:
the point is to get a consistent serialization. 
the Stringer interface is much more intuitive. the object just needs to implement a String() method.
at one point the plan is to replace all FiledNamed to Stringer but that's not a priority.

Reply:
But `Field()` and `String()` might return different textual representations. The former is dedicated to logging and IMO this one should be used.

Reply:
i.cannot think of good use of String() other than logging....